src/describe/sobjectService.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Connection } from '../connection';
+import { DescribeSObjectResult, SObject, SObjectDescribeAPI } from './sobjectApi';
+
+export enum SObjectType {
+  ALL,
+  STANDARD,
+  CUSTOM
+}
+
+export class SObjectService {
+  public readonly connection: Connection;
+
+  public constructor(connection: Connection) {
+    this.connection = connection;
+  }
+
+  public async describeSObject(name: string): Promise<SObject> {
+    const result = await new SObjectDescribeAPI(this.connection).describeSObject(name);
+    return result.result ? Promise.resolve(result.result) : Promise.reject();
+  }
+
+  public async describeSObjects(names: string[]): Promise<SObject[]> {
+    // TODO: make cancellable?
+    const describeAPI = new SObjectDescribeAPI(this.connection);
+    let fetchedResults: DescribeSObjectResult[] = [];
+    let j = 0;
+    while (j < names.length) {
+      try {
+        fetchedResults = fetchedResults.concat(await describeAPI.describeSObjectBatch(names, j));
```

Comment:
@jgellin-sf Would you like to consider parallelizing fetching SObject as part of this effort?

We've found out that it would offer significant performance improvement (26.9s vs 3.6s). Work item W-8035201 has detailed description and link to sample code (just around 30 lines of changes). This is currently under my name but let me know you are interested in taking on it.